src/utils/basic_structs.h

```diff
@@ -0,0 +1,95 @@
+/*!
+ * \file basic_structs.h
+ * \brief Constains some basic structs to be used around the code. Mostly this is here instead of hydro_utilities.h to
+ * avoid circulary dependencies with mhd_utils.h
+ *
+ */
+
+#pragma once
+
+#include "../global/global.h"
+#include "../global/global_cuda.h"
+#include "../utils/gpu.hpp"
+
+namespace hydro_utilities
+{
+// =====================================================================================================================
+// Here are some basic structs that can be used in various places when needed
+// =====================================================================================================================
+/*!
+ * \brief A data only struct that contains the Real members x, y, and z for usage as a vector
+ *
+ */
+struct VectorXYZ {
+  Real x, y, z;
+};
```

Comment:
I haven't looked deeply at this PR (I can, if you want me to --  although I suspect you just want to get done at this point), but I wanted to share a quick thought.

In PR #386, I introduce a class that serves a similar purpose to this struct. I just wanted to share what my class looks like on the off-chance you would want to make use of it here too. But, please don't feel pressured to change anything (especially since it would probably trigger a number of cascading changes on your end). Given the added functionality of my class, I think it would be perfectly fine to have your struct and my functionality coexist. 

The main difference is that my class provides has some additional machinery for supporting index access. For an instance `vec`, the following pairs of expressions are equivalent: ( `vec[0]`, `vec.x()` ); ( `vec[1]`, `vec.y()` ); ( `vec[2]`, `vec.z()` ). While the functionality is a little ugly, the array-access is extremely useful for implementing logic related to positions in for-loops.

```suggestion
struct VectorXYZ {
  /* Tracks the values held by the class. To ensure the class is an aggregate, it needs
   * to be public. With that said, it should be treated as an implementation detail */
  Real arr_[3];

  /* To ensure this class is an aggregate, constructors are implicitly defined.
   * The destructor & move/copy assignment operations are also implicitly defined */

  __device__ __host__ T& operator[](std::size_t i) {return arr_[i];}
  __device__ __host__ const T& operator[](std::size_t i) const {return arr_[i];}
  __device__ __host__ T* data() {return arr_; }

  __device__ __host__ Real& x() noexcept  {return arr_[0];}
  __device__ __host__ const Real& x() const noexcept {return arr_[0];}
  __device__ __host__ Real& y() noexcept  {return arr_[1];}
  __device__ __host__ const Real& y() const noexcept {return arr_[1];}
  __device__ __host__ Real& z() noexcept  {return arr_[2];}
  __device__ __host__ const Real& z() const noexcept {return arr_[2];}
};
```

Again, I'm not really suggesting you make this change. I'm mostly just making you aware that a similar class will exist.

Reply:
I agree that this is better and I'll add it. Thank you!

Reply:
Added it and I made you co-author on the commit.